staging/src/k8s.io/apiserver/pkg/util/applypatch/applypatch.go

```diff
@@ -0,0 +1,257 @@
+package applypatch
```

Comment:
Can we make it clearer that this is test code that shouldn't be used in production?

Reply:
Have you tried to test this test code? :-)

Reply:
Update the file to include a note to use it only for testing


I did try to test with following test snippet

```
package workspace

import (
	"k8s.io/kube-openapi/pkg/util/proto"
	prototesting "k8s.io/kube-openapi/pkg/util/proto/testing"
	"testing"

	apierrors "k8s.io/apimachinery/pkg/api/errors"
	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
	"k8s.io/apimachinery/pkg/runtime/schema"
	"k8s.io/apiserver/pkg/util/applypatch"
	"sigs.k8s.io/yaml"
)

var kubernetesSwaggerSchema = prototesting.Fake{
	Path: "/Users/karthikkn/karthik-temp-k8-workspace/kubernetes/api/openapi-spec/swagger.json",
}

func NewFakeOpenAPIModels() proto.Models {
	d, err := kubernetesSwaggerSchema.OpenAPISchema()
	if err != nil {
		panic(err)
	}
	m, err := proto.NewOpenAPIData(d)
	if err != nil {
		panic(err)
	}
	return m
}

// TestUpdateApplyConflict tests that applying to an object, which
// wasn't created by apply, will give conflicts
func TestUpdateApplyConflict(t *testing.T) {
	m := NewFakeOpenAPIModels()

	f := applypatch.NewDefaultTestFieldManager(m, schema.FromAPIVersionAndKind("apps/v1", "Deployment"))

	patch := []byte(`{
		"apiVersion": "apps/v1",
		"kind": "Deployment",
		"metadata": {
			"name": "deployment",
			"labels": {"app": "nginx"}
		},
		"spec": {
                        "replicas": 3,
                        "selector": {
                                "matchLabels": {
                                         "app": "nginx"
                                }
                        },
                        "template": {
                                "metadata": {
                                        "labels": {
                                                "app": "nginx"
                                        }
                                },
                                "spec": {
				        "containers": [{
					        "name":  "nginx",
					        "image": "nginx:latest"
				        }]
                                }
                        }
		}
	}`)
	newObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
	if err := yaml.Unmarshal(patch, &newObj.Object); err != nil {
		t.Fatalf("error decoding YAML: %v", err)
	}

	if err := f.Update(newObj, "fieldmanager_test"); err != nil {
		t.Fatalf("failed to apply object: %v", err)
	}

	appliedObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
	if err := yaml.Unmarshal([]byte(`{
		"apiVersion": "apps/v1",
		"kind": "Deployment",
		"metadata": {
			"name": "deployment",
		},
		"spec": {
			"replicas": 101,
		}
	}`), &appliedObj.Object); err != nil {
		t.Fatalf("error decoding YAML: %v", err)
	}

	err := f.Apply(appliedObj, "fieldmanager_conflict", false)
	if err == nil || !apierrors.IsConflict(err) {
		t.Fatalf("Expecting to get conflicts but got %v", err)
	}
}

```

Reply:
Why don't you commit the test?
Also, since this is a test package, you should probably name it `applypatch_test.go` or something?

Reply:
I have reused the [test](https://github.com/Karthik-K-N/kubernetes/blob/331525670b772eb8956b7f5204078c51c00aaef3/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go#L196) from the fieldmanager_test.go file thats the reason i have not added that to this commit,

Also the main intention of adding this to util is to make sure we can access function outside scope for example from client_go, If I name it to applypatch_test.go then I will not be able to access the function from this file.

Reply:
Yeah, that's fine, add a test no matter what, even if it's mostly a copy paste from the other location.
I'm not suggesting that you add the `_test.go` but to the package, or maybe actually use `testing` as the package name. Should this code live in https://github.com/kubernetes/kubernetes/tree/331525670b772eb8956b7f5204078c51c00aaef3/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager?

Reply:
Yeah, You are correct, By placing the file over there we can maximize code reuse. Updated it accordingly , Please take a look.